app/src/main/java/com/everest/movieapp/ui/adapters/MovieRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,84 @@
+package com.everest.movieapp.ui.adapters
+
+/**
+ * Recycler view adapter sets the data in view.
+ */
+
+import android.content.Context
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.everest.movieapp.R
+import com.everest.movieapp.data.model.UiMovieDetails
+
+class MovieRecyclerViewAdapter(
+    private var dataModel: List<UiMovieDetails>,
+    private var clickListener: CustomClick
+) :
+    RecyclerView.Adapter<MovieRecyclerViewAdapter.MyViewHolder>() {
+    private var fullMovieList = ArrayList<UiMovieDetails>()
+
+    init {
+        compareChangedData()
```

Comment:
There is no benefit if we use this at first stage, Normally diffutil helps us if there is any change in data